Miscellaneous/docker/full_stack_example/docker-compose-linux.yml

```diff
@@ -0,0 +1,223 @@
+  #
+  #   ELASTIC_VERSION=5.5.1 docker-compose up
+  #
+  # Additionally, the user can control:
+  #   * the total memory assigned to the ES container through the variable ES_MEM_LIMIT e.g. ES_MEM_LIMIT=2g
+  #   * the memory assigned to the ES JVM through the variable ES_JVM_HEAP e.g. ES_JVM_SIZE=1024m
+  #   * the password used for the elastic, logstash_system and kibana accounts through the variable ES_PASSWORD
+  #   * the mysql root password through the var MYSQL_ROOT_PASSWORD
+  #   * the default index pattern used in kibana via the var DEFAULT_INDEX_PATTERN
+  #   * the ES heap size through tt
+  # REF: https://docs.docker.com/compose/compose-file/#variable-substitution
+  #
+
+  #Elasticsearch container
+  elasticsearch:
+    container_name: elasticsearch
+    hostname: elasticsearch
+    image: "docker.elastic.co/elasticsearch/elasticsearch:${ELASTIC_VERSION}"
+    environment:
+      - http.host=0.0.0.0
+      - transport.host=127.0.0.1
+      - bootstrap.memory_lock=true
+      - "ES_JAVA_OPTS=-Xms${ES_JVM_HEAP} -Xmx${ES_JVM_HEAP}"
+    mem_limit: ${ES_MEM_LIMIT}
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+    volumes:
+      - ./config/elasticsearch/elasticsearch.yml:/usr/share/elasticsearch/elasticsearch.yml
+      - esdata:/usr/share/elasticsearch/data
+    #Port 9200 is available on the host
+    ports: ['127.0.0.1:9200:9200']
+    #Healthcheck to confirm availability of ES. Other containers wait on this.
+    healthcheck:
+      test: ["CMD", "curl","-s" ,"-f", "-u", "elastic:${ES_PASSWORD}", "http://localhost:9200/_cat/health"]
+    #Internal network for the containers
+    networks: ['stack']
+
+  #Kibana container
+  kibana:
+    container_name: kibana
+    hostname: kibana
+    image: "docker.elastic.co/kibana/kibana:${ELASTIC_VERSION}"
+    volumes:
+      - ./config/kibana/kibana.yml:/usr/share/kibana/kibana.yml
+    #Port 5601 accessible on the host
+    ports: ['127.0.0.1:5601:5601']
+    networks: ['stack']
+    #We don't start Kibana until the ES instance is ready
+    depends_on: ['elasticsearch']
+    environment:
+      - "ELASTICSEARCH_PASSWORD=${ES_PASSWORD}"
+    healthcheck:
+      test: ["CMD", "curl", "-s", "-f", "http://localhost:5601/login"]
+      retries: 6
+
+  #Heartbeat container
+  heartbeat:
+    container_name: heartbeat
+    hostname: heartbeat
+    image: "docker.elastic.co/beats/heartbeat:${ELASTIC_VERSION}"
+    volumes:
+      #Mount the heartbeat configuration so users can make edits
+      - ./config/beats/heartbeat/heartbeat.yml:/usr/share/heartbeat/heartbeat.yml
+    depends_on:
+      elasticsearch:  { condition: service_healthy }
+    environment:
+      - "ES_PASSWORD=${ES_PASSWORD}"
+    command: heartbeat -e -E output.elasticsearch.username=elastic -E output.elasticsearch.password=${ES_PASSWORD}
+    networks: ['stack']
+    restart: on-failure
+
+  #Filebeat container
+  filebeat:
+    container_name: filebeat
+    hostname: filebeat
+    user: root
```

Comment:
we have to access the docker container logs unfortunately. I'll document in the README